lib/irb/source_finder.rb

```diff
@@ -43,17 +43,18 @@ def find_source(signature)
     private
 
     def find_end(file, first_line)
-      lex = RubyLex.new(@irb_context)
+      local_variables = @irb_context.local_variables
```

Comment:
I think local_variables should be `[]` here.
Parsing a file does not need to depend on binding's local variable.


```ruby
# a.rb
a /%#/ if false; def foo;end #.size; def foo;"

binding.irb
__END__
"
end
```
```
$ ruby -Ilib a.rb
irb(main):001> $ foo
# shows source of foo
irb(main):002> a = 1
=> 1
irb(main):003> $ foo
# shows wrong source range of foo
```

Reply:
Good point. Updated 👍